src/scripts/main.js

```diff
@@ -1,3 +1,57 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const thList = document.querySelectorAll('th');
+
+thList.forEach((th) => {
+  th.dataset.name = th.innerHTML.toLowerCase();
+});
+
+table.addEventListener('click', (e) => {
+  if (e.target.tagName !== 'TH') {
+    return;
+  }
+
+  const th = e.target;
+  const thIndex = th.cellIndex;
+  const thName = th.dataset.name;
+
+  sortTable(thIndex, thName);
+});
+
+function sortTable(colNum, type) {
+  const tbody = document.querySelector('tbody');
+  const rowsArray = Array.from(tbody.rows);
+
+  switch (type) {
```

Comment:
As the best practice, the `switch/case` block should have a `default` statement. It may return `null` or some default value.